traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -821,14 +821,19 @@ func readGetDeliveryServices(params map[string]string, tx *sqlx.Tx, user *auth.C
 
 	where, orderBy, pagination, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(params, queryParamsToSQLCols)
 	if len(errs) > 0 {
-		return nil, errs, tc.DataConflictError
+		for _, err := range errs {
+			if err.Error() == `id cannot parse to integer` { // TODO create const for string
```

Comment:
why the addition here?  Also why does an id not being able to be parsed to an int mean not found?

FWIW In our generic read, that is used across many handlers, we just join the errors and mark it is a bad request. Should we follow that here?

Reply:
@mhoppa I saw "matches perl response" and didn't immediately want to break BC. I agree that marking it as bad request makes more sense.

More info:

`TODeliveryService.Read()` has returned `tc.DataMissingError` for that case [since deliveryservices's original Go implementation](https://github.com/apache/trafficcontrol/blob/75a949dedb/traffic_ops/traffic_ops_golang/deliveryservice/deliveryservicesv13.go#L280-L284) in 75a949dedb. Because `readGetDeliveryServices()`'s return values no longer include `[]error`, moving the logic from `TODeliveryService.Read()` to `readGetDeliveryServices()` is one way to preserve that response.

The original Perl module [did not distinguish](https://github.com/apache/trafficcontrol/blob/55ff2567d1/traffic_ops/app/lib/API/DeliveryService.pm#L53-L55) between not finding a delivery service and the id param failing int validation. So, even though that behavior was [originally documented](https://github.com/apache/trafficcontrol/blob/55ff2567d1/traffic_ops/doc/api_response_examples.txt#L165-L171), now that we have type safety, `http.StatusBadRequest` makes more sense.

Reply:
Addressed in ff7334ac25.